shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java

```diff
@@ -1334,10 +1331,19 @@ public static class HdfsEncryptionShim implements HadoopShims.HdfsEncryptionShim
 
     private final Configuration conf;
 
-    public HdfsEncryptionShim(URI uri, Configuration conf) throws IOException {
+    public static HdfsEncryptionShim createInstance(URI uri, Configuration conf) throws IOException {
+      HdfsAdmin hadmin;
+      KeyProvider keyP;
+      hadmin = new HdfsAdmin(uri, conf);
+      keyP = hadmin.getKeyProvider();
```

Comment:
```suggestion
      HdfsAdmin hadmin = new HdfsAdmin(uri, conf);
      KeyProvider keyP = hadmin.getKeyProvider();
```

Thanks for the update! In this case, we may not have to separate assignments from declarations.

Reply:
I didn't see that, I'll update

Reply:
done updated 👍